File: plugin/testData/shared/basics/When.shared.kt
package com.test

fun main() {
    var number: Int = 0
    var value = 43.toInt()
    when (value) {
        1 -> number += 1
        2 -> {
            number += 2
            number += 3
        }
        43 -> {
            number += 4
        }
        else -> {
            number -= 99
        }
    }
    when {
        value == 42 -> number += 8
        value == 41 -> {
            number += 16
        }
        value == 43 -> number += 32
        else -> {
        }
    }
    println(number)

    println(
        when (value) {
            0 -> "Hi"
            else -> {
                "Nope"
            }
        }
    )
}

File: plugin/testDataOutputTs/shared/basics/When.shared.ts
//Package: com.test
//Converted using Khrysalis2


export function main(): void {
    let aNumber : number = (0);
    let value = (43).toInt();
    switch(value) {
        case (1):
        aNumber += (1);
        break;
        case (2):
        aNumber += (2);
        aNumber += (3);
        break;
        case (43):
        aNumber += (4);
        break;
        default:
        aNumber -= (99);
        break;
    };
    if (value == (42)) {
        aNumber += (8);
    }else if (value == (41)) {
        aNumber += (16);
    }else if (value == (43) || value == (44)) {
        aNumber += (32);
    }else { 
    };
    console.log(aNumber);
    aNumber = (() => {
            if (value == (42)){
                return (8);
            }else if (value == (41)){
                return (16);
            }else if (value == (43) || value == (44)){
                return (32);
            }else {
                return (0);
            }
    })();
    console.log(aNumber);
    console.log((() => {
                switch(value) {
                    case (0):
                    return "Hi";
                    default:
                    return "Nope";
                }
    })())
}

